tests/speculos/test_eip191.py

```diff
@@ -0,0 +1,209 @@
+            cmd.client.press_and_release('right')
+
+            # Message
+            compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_metamask/00001.png")
+            cmd.client.press_and_release('right')
+
+            # Sign message
+            compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_metamask/00002.png")
+            cmd.client.press_and_release('both')
+
+    v, r, s = result
+
+    assert v == 0x1c # 28
+    assert r.hex() == "916099cf0d9c21911c85f0770a47a9696a8189e78c259cf099749748c507baae"
+    assert s.hex() == "0d72234bc0ac2e94c5f7a5f4f9cd8610a52be4ea55515a85b9703f1bb158415c"
+ 
+def test_personal_sign_reject(cmd):
+    result: list = []
+
+    bip32_path="44'/60'/0'/0/0"
+    transaction = PersonalTransaction(
+        msg="This is an reject sign"
+    )
+
+    try:
+        with cmd.personal_sign_tx(bip32_path=bip32_path, transaction=transaction, result=result) as ex:
+            sleep(0.5)
+
+            if cmd.model == "nanos":
+                # Sign message
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00000.png")
+                cmd.client.press_and_release('right')
+
+                # Message 1/2, 2/2
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00001.png")
+                cmd.client.press_and_release('right')
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00002.png")
+                cmd.client.press_and_release('right')
+
+                # Sign message
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00003.png")
+                cmd.client.press_and_release('right')
+
+                # Cancel signature
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00004.png")
+                cmd.client.press_and_release('both')
+
+            if cmd.model == "nanox" or cmd.model == "nanosp":
+                # Sign message
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00000.png")
+                cmd.client.press_and_release('right')
+
+                # Message
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00001.png")
+                cmd.client.press_and_release('right')
+
+                # Sign message
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00002.png")
+                cmd.client.press_and_release('right')
+
+                # Cancel signature
+                compare_screenshot(cmd, f"screenshots/eip191/{PATH_IMG[cmd.model]}/personal_sign_reject/00003.png")
+                cmd.client.press_and_release('both')
+
+    except ethereum_client.exception.errors.DenyError as error:
```

Comment:
If the error is expected, rather than a `try ... except ...` which does not guarantee that the exception is raised (test will pass weither the code raises or not), use the `pytest.raises` method:

```python
with pytest.raises(ethereum_client.exception.errors.DenyError) as error:
   [ your code ]
assert error.value.args[0] == '0x6985'
```

Reply:
https://github.com/LedgerHQ/app-ethereum/pull/275/commits/8da09ec892a6bdabf622f3e0c024ccec06094a81

Reply:
it's much clearer :)